spec/system/candidate_interface/candidate_is_only_presented_with_valid_radio_buttons_when_replacing_a_course_choice_spec.rb

```diff
@@ -0,0 +1,161 @@
+
+    when_i_arrive_at_my_application_dashboard
+    and_i_click_update_my_course_choice
+    then_i_arrive_on_the_replace_course_choice_page
+    and_i_should_be_told_the_course_is_full
+    and_i_should_have_the_option_to_replace_my_course_choice
+    and_i_should_have_the_option_to_delete_my_course_choice
+    and_i_should_have_the_option_to_keep_my_course_choice
+    and_i_should_not_have_the_option_to_change_the_location_of_my_course_choice
+    and_i_should_not_have_the_option_to_replace_the_study_mode_course_choice
+
+    given_that_there_is_another_location_and_study_mode_available
+
+    when_i_arrive_at_my_application_dashboard
+    and_i_click_update_my_course_choice
+    then_i_arrive_on_the_replace_course_choice_page
+    and_i_should_be_told_there_are_no_more_full_time_places_at_my_location
+    and_i_should_have_the_option_to_replace_my_course_choice
+    and_i_should_have_the_option_to_delete_my_course_choice
+    and_i_should_have_the_option_to_keep_my_course_choice
+    and_i_should_have_the_option_to_change_the_location_of_my_course_choice
+    and_i_should_have_the_option_to_replace_the_study_mode_course_choice
+
+    given_the_other_study_mode_becomes_full
+
+    when_i_arrive_at_my_application_dashboard
+    and_i_click_update_my_course_choice
+    then_i_arrive_on_the_replace_course_choice_page
+    and_i_should_be_told_there_are_no_more_places_at_that_location_available
+    and_i_should_have_the_option_to_replace_my_course_choice
+    and_i_should_have_the_option_to_delete_my_course_choice
+    and_i_should_have_the_option_to_keep_my_course_choice
+    and_i_should_have_the_option_to_change_the_location_of_my_course_choice
+    and_i_should_not_have_the_option_to_replace_the_study_mode_course_choice
+
+    given_that_part_time_study_mode_becomes_available_and_all_other_locations_become_full
+
+    when_i_arrive_at_my_application_dashboard
+    and_i_click_update_my_course_choice
+    then_i_arrive_on_the_replace_course_choice_page
+    and_i_should_be_told_there_are_no_more_full_time_places_available
+    and_i_should_have_the_option_to_replace_my_course_choice
+    and_i_should_have_the_option_to_delete_my_course_choice
+    and_i_should_have_the_option_to_keep_my_course_choice
+    and_i_should_not_have_the_option_to_change_the_location_of_my_course_choice
+    and_i_should_have_the_option_to_replace_the_study_mode_course_choice
+
+    given_the_course_has_been_withdrawn
+
+    when_i_arrive_at_my_application_dashboard
+    and_i_click_update_my_course_choice
+    then_i_arrive_on_the_replace_course_choice_page
+    and_i_should_be_told_that_the_course_has_been_withdrawn
+    and_i_should_have_the_option_to_replace_my_course_choice
+    and_i_should_have_the_option_to_delete_my_course_choice
+    and_i_should_have_the_option_to_keep_my_course_choice
+    and_i_should_not_have_the_option_to_change_the_location_of_my_course_choice
+    and_i_should_not_have_the_option_to_replace_the_study_mode_course_choice
+  end
```

Comment:
Would it have been easier to test this using a component and multiple unit tests? This seems like quite a lot of effort!

Reply:
It's largely just chaining the same methods over and over uselessly tbh. But yea i think I'm going to put it into a component anyway actually. Just so i can pull all that logic out of the controller if nothing else.